coralproject/cay
# E2E Test for Max and Min Char Limit in Short and Long Answers

## What does this PR do?

This PR adds the E2E tests for the Max and Min Char limit in Short and Long Answers (TextField, TextArea) and enables them back
- Adds missing selectors
- Provides an API for testing the `createFormPage` and the `standAloneFormPage`
  
  `createFormPage` : The page where the user creates forms
  `standAlonePage` : The page that renders the stand alone form

_Thoughts and TODOs:_
- Tests are going to be big, maybe its best if we separate this into tiny modules (i.e: max min, required field, etc)
- The tests for short and long answer are almost the same. Maybe we can trim code by knowing this.
- Nightwatch setValue **doesn't** set the value. It appends the new value. [Bug Issue](https://github.com/nightwatchjs/nightwatch/issues/4)
  That's why I had to use clearValue 
- If the test fails at first it may has something to do with network issues, and the server not responding back quickly. Try again and contact me.
## How do I test this PR?
- Checkout the branch `max-min-tests` in Cay
  
  ``` bash
  git checkout max-min-tests
  ```
- Checkout the branch `max-min-issues` in Elkhorn
  
  ``` bash
  git checkout max-min-issues
  ```
- Run the tests 
  
  ``` bash
  npm run e2e
  ```
- Check if Min and Max Char limit is working as expected.

@coralproject/frontend
